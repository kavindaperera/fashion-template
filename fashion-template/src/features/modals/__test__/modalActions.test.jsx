import configureStore from "redux-mock-store";
// Actions to be tested
import * as selectActions from "../modalActions";

const mockStore = configureStore();
const store = mockStore();

describe("openModal", () => {
  beforeEach(() => {
    // Runs before each test in the suite
    store.clearActions();
  });
  test("Dispatches the correct action and payload", () => {
    const expectedActions = [
      {
        type: "MODAL_OPEN",
        payload: {
          modalType: undefined,
          modalProps: undefined,
        },
      },
    ];
    store.dispatch(selectActions.openModal());
    expect(store.getActions()).toEqual(expectedActions);
  });
});


describe("closeModal", () => {
    beforeEach(() => {
      // Runs before each test in the suite
      store.clearActions();
    });
    test("Dispatches the correct action and payload", () => {
      const expectedActions = [
        {
          type: "MODAL_CLOSE",
        },
      ];
      store.dispatch(selectActions.closeModal());
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

   /*SnapShot Testing*/ 


describe("openModal", () => {
    beforeEach(() => {
      // Runs before each test in the suite
      store.clearActions();
    });
    test("Dispatches the correct action and payload", () => {
      store.dispatch(selectActions.openModal());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

describe("closeModal", () => {
    beforeEach(() => {
      // Runs before each test in the suite
      store.clearActions();
    });
    test("Dispatches the correct action and payload", () => {
      store.dispatch(selectActions.closeModal());
      expect(store.getActions()).toMatchSnapshot();
    });
  });