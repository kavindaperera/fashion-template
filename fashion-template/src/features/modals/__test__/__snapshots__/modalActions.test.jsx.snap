// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`closeModal Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "MODAL_CLOSE",
  },
]
`;

exports[`openModal Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "modalProps": undefined,
      "modalType": undefined,
    },
    "type": "MODAL_OPEN",
  },
]
`;
